walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,57 @@
 
 package com.google.sps;
 
+import java.lang.Math;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.PriorityQueue;
 
+/**
+ * Returns a Collection of TimeRange objects that represents the available time slots for a meeting
+ * to hold
+ *
+ * @param events     a collection of other events happening on the day of the meeting
+ * @param request    a meeting request with a duration and a collection of attendees
+ * @return           a collection of TimeRanges in which the meeting could hold
+ */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    PriorityQueue<Event> eventQueue = new PriorityQueue<>(Event.ORDER_BY_START_TIME);
+    eventQueue.addAll(events);
+
+    Collection<TimeRange> availableTimes = new ArrayList<TimeRange>();
+    int startOfAvailableTime = TimeRange.START_OF_DAY;
+
+    while (!eventQueue.isEmpty()) {
+      Event event = eventQueue.remove();
+      int eventStart = event.getWhen().start();
+      int eventEnd = event.getWhen().end();
+
+      if (hasAttendees(event, request.getAttendees())) {
```

Comment:
FYI (no need to change): it would be slightly clearer to invert this check: if the event does not have attendees in the intersection, explicitly skip it. You could put this check immediately after popping the queue, but before anything else (i.e. before declaring `eventStart`). Ultimately, this is personal preference, however.